prog/postgres/converge_postgres_resource.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+require "forwardable"
+
+require_relative "../../lib/util"
+
+class Prog::Postgres::ConvergePostgresResource < Prog::Base
+  subject_is :postgres_resource
+
+  label def provision_servers
+    hop_wait_servers_to_be_ready if postgres_resource.has_enough_fresh_servers?
+
+    if postgres_resource.servers.none? { _1.vm.vm_host.nil? }
+      exclude_host_ids = (Config.development? || Config.is_e2e) ? [] : postgres_resource.servers.map { _1.vm.vm_host.id }
+      Prog::Postgres::PostgresServerNexus.assemble(resource_id: postgres_resource.id, timeline_id: postgres_resource.timeline.id, timeline_access: "fetch", exclude_host_ids: exclude_host_ids)
+    end
+
+    nap 5
+  end
+
+  label def wait_servers_to_be_ready
+    hop_provision_servers unless postgres_resource.has_enough_fresh_servers?
+    hop_recycle_representative_server if postgres_resource.has_enough_ready_servers?
+
+    nap 60
+  end
+
+  label def recycle_representative_server
+    if postgres_resource.representative_server
```

Comment:
How can a postgres resource not have a representative server? During the failover maybe?

Reply:
Yes, and this label itself might trigger a failover in case the representative server needs to be recycled so we wait here until a new one comes up.

Reply:
Yes, during failover, there is a time window without a representative server.

Actually, while preparing this PR, I realized that there are many places where we assume representative server exist, which is not the case all the time. Thankfully, assuming its existence would not cause any permanent prblem. It only causes error to be thrown and we retry in next iteration (and eventually failover completes and we get a representative server). Still, it would be better to not assume. I put that down as a repair item for me.